tsdb/index/inmem/inmem.go

```diff
@@ -189,13 +189,23 @@ func (i *Index) MeasurementIterator() (tsdb.MeasurementIterator, error) {
 func (i *Index) CreateSeriesListIfNotExists(seriesIDSet *tsdb.SeriesIDSet, measurements map[string]int,
 	keys, names [][]byte, tagsSlice []models.Tags, opt *tsdb.EngineOptions, ignoreLimits bool) error {
 
+	// Verify that the series will not exceed limit.
+	if !ignoreLimits {
+		i.mu.RLock()
+		if max := opt.Config.MaxSeriesPerDatabase; max > 0 && len(i.series)+len(keys) > max {
```

Comment:
i like simplifying `opt.Config.MaxSeriesPerDatabase` to `max`.  i feel like similarly simplifying `len(i.series)+len(keys)` would make the expression simpler. so instead of:
```
if max := opt.Config.MaxSeriesPerDatabase; max > 0 && len(i.series)+len(keys) > max {
```

maybe:

```
if max, numSeries := opt.Config.MaxSeriesPerDatabase, len(i.series)+len(keys); max > 0 && numSeries > max {
```
i feel like that clarifies the expression.  i'm not sure if `numSeries` is the right name for it but that's the gist.

Reply:
Initially, I didn't have the `RLock()` here because the code was moved up from a later part of the function where it had been inside a lock. The race detector caught the race and I added the read lock here. I re-ran the tests multiple times with no race detected. I know the race detector can only report races that actually occur (i.e., no false positives but can have false negatives). I don't think there's a race in the sense that two goroutines might simultaneously access `i.series` but there is a known condition that two or more simultaneous writes might make it past this check and put the database past the limit. The limit isn't an exact limit. We sacrifice exact limit enforcement for performance.

I agree that we could cleanup those bits of code but this is a backport so it's best not to make cosmetic changes here. We should make them on the `1.8` branch first  and then backport.